zerver/lib/i18n.py

```diff
@@ -67,3 +68,18 @@ def get_and_set_request_language(
     RequestNotes.get_notes(request).set_language = translation.get_language()
 
     return request_language
+
+
+def get_browser_locale(request: HttpRequest) -> Optional[str]:
+    accept_lang_header = request.META.get("HTTP_ACCEPT_LANGUAGE")
+    if accept_lang_header is None:
+        return None
+
+    available_language_codes = get_available_language_codes()
+    for accept_lang, priority in parse_accept_lang_header(request.META.get("HTTP_ACCEPT_LANGUAGE")):
+        if accept_lang == "*":
+            return None
```

Comment:
Not sure what to do in this case when the header is set to `Any language` (though I am not sure how can I reproduce this in a browser). For now, this returns None which means realm-level default is used.